src/openrct2/paint/VirtualFloor.cpp

```diff
@@ -177,9 +175,9 @@ bool virtual_floor_tile_is_floor(int16_t x, int16_t y)
 
     // Check if map selection (usually single tiles) are enabled
     //  and if the current tile is near or on them
-    if ((gMapSelectFlags & MAP_SELECT_FLAG_ENABLE) && x >= gMapSelectPositionA.x - _virtualFloorBaseSize
-        && y >= gMapSelectPositionA.y - _virtualFloorBaseSize && x <= gMapSelectPositionB.x + _virtualFloorBaseSize
-        && y <= gMapSelectPositionB.y + _virtualFloorBaseSize)
+    if ((gMapSelectFlags & MAP_SELECT_FLAG_ENABLE) && loc.x >= gMapSelectPositionA.x - _virtualFloorBaseSize
+        && loc.y >= gMapSelectPositionA.y - _virtualFloorBaseSize && loc.x <= gMapSelectPositionB.x + _virtualFloorBaseSize
+        && loc.y <= gMapSelectPositionB.y + _virtualFloorBaseSize)
```

Comment:
Would it look better like:
```c++
    const auto virtualFloorBaseCoords = CoordsXY{_virtualFloorBaseSize, _virtualFloorBaseSize};
    if ((gMapSelectFlags & MAP_SELECT_FLAG_ENABLE) && loc >= (gMapSelectPositionA - virtualFloorBaseCoords)
        && loc <= (gMapSelectPositionB + virtualFloorBaseCoords)
```
I'm unsure if putting it into `CoordsXY` are a bit of stretch

Reply:
Another option would be to overload the CoordsXY + operator to also allow integers. But I don't think that should go into this PR, if I'm honest.

Reply:
I'm fine with it being done on another pr

Reply:
What would a + overload mean? Add integer to x coordinate, add to both x and y? It's a little ambiguous. I'm not 100% sure if the <= overloads make great sense either. What does it mean for a coordinate to be less than another.

Reply:
I'm with @duncanspumpkin, it adds some ambiguity, if theres some distance measurement involved it should be its own distance function outside of the class or make it a static to accept A and B as parameters.